### DISCUSSION TITLE
Why is `i18n` disabled in `output: 'export'`?

### DISCUSSION THREAD

**Discussion by JoshBashed** [vercel/next.js#70471] [Help]
### Summary

I want to build an app on Azure Static Sites. This requires static export. I also intend to localize my website with the i18n feature. Unfortunately, there is a restriction that disables this.

I am struggling to understand why this restriction is in place. According to this [StackOverFlow](https://stackoverflow.com/questions/77921083/how-can-ssgi18ndynamic-routes-be-managed-in-nextjs-14-strictly-using-the-pag) thread. The functionality works just fine. Why can't this just be a warning in that case? Why does it need a workaround?

**Comment by icyJoseph** [comment]
I think the restriction is in place, because of redirects. 

Since the i18n object, as far as I remember, is a pages router thing, going to the docs for static exports on Pages Router, has:

```md
[Unsupported Features](https://nextjs.org/docs/pages/building-your-application/deploying/static-exports#unsupported-features)
Features that require a Node.js server, or dynamic logic that cannot be computed during the build process, are not supported:

[Internationalized Routing](https://nextjs.org/docs/pages/building-your-application/routing/internationalization)
```

- https://nextjs.org/docs/pages/building-your-application/deploying/static-exports#unsupported-features
- https://nextjs.org/docs/pages/building-your-application/routing/internationalization

The last link, lists a couple of cases, on which the server would do redirects, and such, and I think that's the root for static exports, not supporting i18n.

**Reply by JoshBashed** [reply]
I mean, i don't think it should matter cause all the language codes are known at compile time.

**Reply by icyJoseph** [reply]
🤷‍♂️ the problem is that the feature also allows for, redirecting from some locales to others, for example those not generated during build time, and sure, we could say, just do a 404, for export mode, or use a fallback, but that's not implemented, and its just a guess. The root issue being that a server is needed to do redirects properly, and that leaves out export mode.

**Comment by landsman** [comment]
I would love to see support for this.
